core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -162,6 +163,34 @@ public String defaultHostname() {
         }
     }
 
+    /**
+     * Returns the local {@link ServerPort} that this {@link Server} is listening to.
+     *
+     * @throws IllegalStateException if there is no active local port available
+     *                               or the server is not started yet
+     */
+    public int activeLocalPort() {
+        return activeLocalPort(null);
+    }
+
+    /**
+     * Returns the local {@link ServerPort} which serves the given {@link SessionProtocol}.
+     *
+     * @throws IllegalStateException if there is no active local port available for the given
+     *                               {@link SessionProtocol} or the server is not started yet
+     */
+    public int activeLocalPort(@Nullable SessionProtocol protocol) {
```

Comment:
Could you make the parameter `protocol` non-null? You could create an internal method `activeLocalPort0(@Nullable SessionProtocol)`:

```java
public int activeLocalPort() {
    return activeLocalPort0(null);
}

public int activeLocalPort(SessionProtocol protocol) {
    requireNonNull(protocol, "protocol");
    return activeLocalPort0(protocol);
}

private int activeLocalPort(@Nullable SessionProtocol protocol) { ... }
```